tests/components/matter/test_light.py

```diff
@@ -14,55 +15,185 @@
 )
 
 
-@pytest.fixture(name="light_node")
-async def light_node_fixture(
+@pytest.fixture(name="on_off_light_node")
+async def on_off_light_node_fixture(
     hass: HomeAssistant, matter_client: MagicMock
 ) -> MatterNode:
-    """Fixture for a light node."""
+    """Fixture for an on/off light node."""
+    return await setup_integration_with_node_fixture(hass, "onoff-light", matter_client)
+
+
+@pytest.fixture(name="dimmable_light_node")
+async def dimmable_light_node_fixture(
+    hass: HomeAssistant, matter_client: MagicMock
+) -> MatterNode:
+    """Fixture for a dimmable light node."""
+    return await setup_integration_with_node_fixture(
+        hass, "dimmable-light", matter_client
+    )
+
+
+@pytest.fixture(name="color_temperature_light_node")
+async def color_temperature_light_node_fixture(
+    hass: HomeAssistant, matter_client: MagicMock
+) -> MatterNode:
+    """Fixture for a color temperature light node."""
+    return await setup_integration_with_node_fixture(
+        hass, "color-temperature-light", matter_client
+    )
+
+
+@pytest.fixture(name="extended_color_light_node")
+async def extended_color_light_node_fixture(
+    hass: HomeAssistant, matter_client: MagicMock
+) -> MatterNode:
+    """Fixture for an extended color light node."""
     return await setup_integration_with_node_fixture(
         hass, "extended-color-light", matter_client
     )
 
 
-async def test_turn_on(
+async def test_on_off_light(
     hass: HomeAssistant,
     matter_client: MagicMock,
-    light_node: MatterNode,
+    on_off_light_node: MatterNode,
```

Comment:
Instead of using this fixture, I suggest we use the `pytest.mark.parametrize` decorator and add a parameter for the fixture name, eg `"onoff-light"`, `"dimmable-light"` etc. Then use `setup_integration_with_node_fixture` either inside the test or inside an `indirect` fixture to load the fixture file and set up the integration. This way we can test the on/off features for all light devices that support on/off with a single test code.

https://docs.pytest.org/en/latest/how-to/parametrize.html

To make the tests work, we'll need to add more parameters as needed, eg one for `entity_id` for each fixture.